pkg/services/datasources/models.go

```diff
@@ -70,6 +70,16 @@ type DataSource struct {
 
 	Created time.Time `json:"created,omitempty"`
 	Updated time.Time `json:"updated,omitempty"`
+
+	isSecureSocksDSProxyEnabled *bool `xorm:"-"`
+}
+
+func (ds *DataSource) IsSecureSocksDSProxyEnabled() bool {
```

Comment:
Should we update [here](https://github.com/grafana/grafana/blob/main/pkg/services/datasources/service/datasource.go#L748) to use this too?